### PR TITLE
feat: create new branch from branches page (#162)

### LINKED ISSUE #162: Create branch from branches page
We have a feature to create branch from project detail page but not from branches page for the particular project currently.
image

### PR DESCRIPTION
**Pull request by ghost**
This PR will add feature to add branch from branches page

 Closes #162

### CODE REVIEW

File: components/branches/CreateBranchModal.tsx [hunk 1]
```diff
@@ -26,9 +26,14 @@ interface Project {
...
   isOpen: boolean;
   setIsOpen: Dispatch<SetStateAction<boolean>>;
+  onSuccessCreation: () => void;
 }
 
-const CreateBranchModal = ({ isOpen, setIsOpen }: BranchModalProps) => {
+const CreateBranchModal = ({
+  isOpen,
+  setIsOpen,
+  onSuccessCreation,
```
Comment (dahal) [inline components/branches/CreateBranchModal.tsx:35]: Lets rename to `onSuccess` , const will self define that you are creating something.

File: pages/projects/[id]/branches.tsx [hunk 2]
```diff
@@ -180,6 +183,14 @@ export const BranchesPage = ({ projects, currentProject }: Props) => {
       projects={projects}
       currentProject={currentProject}
     >
+      <CreateBranchModal
+        onSuccessCreation={() => {
+          utils.branches.getAll.invalidate();
```
Comment (dahal) [inline pages/projects/[id]/branches.tsx:188]: Curious what `invalidate` will do, i have not checked doc yet, asking for myself to learn more about trpc.

**Review by dahal — CHANGES_REQUESTED** [review]
LGTM, lets fix the onSuccess and we are good to go 🚀

File: components/branches/CreateBranchModal.tsx (hunk 1 above)
Comment (dahal) [reply components/branches/CreateBranchModal.tsx:35]: image

File: pages/projects/[id]/branches.tsx (hunk 2 above)
Comment (dahal) [reply pages/projects/[id]/branches.tsx:188]: image

Comment (chetannn) [reply pages/projects/[id]/branches.tsx:188]: `invaidate` will invalidate current query data and refetch fresh records

**Pull request merged**